lib/src/file_context.dart

```diff
@@ -0,0 +1,65 @@
+import 'dart:io';
+
+import 'package:analyzer/dart/analysis/analysis_context_collection.dart';
+import 'package:analyzer/dart/analysis/results.dart';
+import 'package:analyzer/dart/analysis/utilities.dart';
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:path/path.dart' as p;
+import 'package:source_span/source_span.dart';
+
+import 'patch.dart';
+
+/// A helper class for a file located at [path] that provides access to its
+/// contents and analyzed formats like [CompilationUnit] and [LibraryElement].
+class FileContext {
```

Comment:
One of the main changes in this PR: adding a `FileContext` wrapper around each file and passing this to `Suggestor.generatePatches` instead of a `SourceFile`.

A downside of this change is that it makes suggestors slightly harder to test because you have to create a `FileContext` instead of just a `SourceFile.fromString()`, but we can still make that easy by providing a util to accomplish the same thing.

The benefit is that it provides a single place to get information about the file in various formats (raw source text, `SourceFile`, unresolved `CompilationUnit`, resolved `CompilationUnit`, or resolved `LibraryElement`), and does so without the suggestor needing to be responsible for any additional setup like creating the analysis context.